polars/polars-lazy/polars-plan/src/dsl/function_expr/datetime.rs

```diff
@@ -146,10 +146,7 @@ pub(super) fn cast_timezone(s: &Series, tz: &str) -> PolarsResult<Series> {
 pub(super) fn tz_localize(s: &Series, tz: &str) -> PolarsResult<Series> {
     let ca = s.datetime()?.clone();
     match ca.time_zone() {
-        Some(tz) if tz == "UTC" => {
-           Ok(ca.with_time_zone(Some("UTC".into()))?.into_series())
-        }
```

Comment:
This was added in https://github.com/pola-rs/polars/pull/5522/files , but it looks like it's no longer necessary

Reply:
Hmm maybe the intention was to match the target tz? I'll update later, marking as draft for now

Reply:
I've rewritten according to what I think the intention was